### ISSUE TITLE
[dlangui Example1] Window Performance problems in Windows 10

### ISSUE THREAD

**Issue by BoQsc** [buggins/dlangui#614]
https://user-images.githubusercontent.com/21064622/125163796-80136c80-e197-11eb-8fc1-ffb2b9ba4461.mp4

**Comment by GrimMaple** [comment]
I tested it on the most recent commit and didn't see any jittering. Is this still an issue?

**Comment by Superbelko** [comment]
I think this is only noticeable in debug builds, optimized builds using LDC is fast enough even on 4k resolution.

**Comment by BoQsc** [comment]
The performance issues might only be noticible on a lower-end computer.

**Comment by Superbelko** [comment]
Yeah, but for me it looks more like an layout issue (also flicker issue) rather than performance issue, like as if it is doing wrong measurements and then constantly overshoot desired sizes in both ways causing this ugly visual jitter.

Anyway, "example1" is more like stress test, in more simpler layouts it should be much more performant.

**Comment by Superbelko** [comment]
It seems OpenGL implementation is buggy, for now as a workaround it can be avoided using "minimal" configuration that draws using system graphics instead of OpenGL, using it provides much better CPU utilization (but higher CPU load due to drawing using CPU) and performance (in release builds) without such hiccups.

Here is how to add "minimal" configuration to a project, in dub.json add "configurations" section like this:
https://github.com/buggins/dlangui/blob/d92f668fbefc0ddbcef4b700514a3b0906ffd86b/examples/example1/dub.json#L41-L46

And this is how to build it using for example with LDC compiler:
`dub build -c minimal --compiler=ldc2 --build release`

**Comment by GrimMaple** [comment]
> It seems OpenGL implementation is buggy, for now as a workaround it can be avoided using "minimal" configuration that draws using system graphics instead of OpenGL, using it provides much better CPU utilization (but higher CPU load due to drawing using CPU) and performance (in release builds) without such hiccups.
> 
> Here is how to add "minimal" configuration to a project, in dub.json add "configurations" section like this:
> 
> https://github.com/buggins/dlangui/blob/d92f668fbefc0ddbcef4b700514a3b0906ffd86b/examples/example1/dub.json#L41-L46
> 
> And this is how to build it using for example with LDC compiler: `dub build -c minimal --compiler=ldc2 --build release`

This isn't an OpenGL bug, it's a Windows bug. `SwapBuffers` on Windows is amazingly slow, and must come along with `glFlush` AND no vsync enabled. I've had the same issue with my game engine.

I'll take a look at this soon, and try to figure out a solution

**Comment by Superbelko** [comment]
Makes sense, on Windows all UI is displayed by compositor since Vista, waiting for monitor vsync no longer an option.

So I've quickly googled the problem (esp. chromium discussion about it) and the solution usually involves relying either on wait for compositor after swap using `DwmFlush()` (didn't worked for me though) or implementing high frequency timer for smooth framerate.
